lib/APPTurtle/src/Subscriber.h

```diff
@@ -0,0 +1,255 @@
+};
+
+/**
+ * Subscriber class for an specific message type.
+ * @tparam messageType ROS2 Topic message type
+ */
+template<typename messageType>
+class Subscriber : public BaseSubscriber
+{
+public:
+    /**
+     * ROS Topic Prototype Callback.
+     * Provides the received data in the respective topic to the application.
+     *
+     * @param[in] msgData       Received data.
+     */
+    typedef std::function<void(const messageType* msgData)> RosTopicCallback;
+
+    /**
+     * Construct a subscriber.
+     *
+     * @param[in] topicName     Name of the topic to subscribe to.
+     * @param[in] typeSupport   Pointer to support structure for the message type.
+     * @param[in] callback      Callback for the topic subscriber.
+     */
+    Subscriber(const String& topicName, const rosidl_message_type_support_t* typeSupport, RosTopicCallback callback) :
+        BaseSubscriber(topicName, typeSupport),
+        m_callback(callback),
+        m_buffer()
+    {
+    }
+
+    /**
+     * Default destructor.
+     */
+    virtual ~Subscriber()
+    {
+    }
+
+    /**
+     * Initialize subscription.
+     *
+     * @param[in] node      Pointer to the RCL node.
+     * @param[in] executor  Pointer to the RCLC executor.
+     *
+     * @returns If succesfully initialized, returns true. Otherwise, false.
+     */
+    bool init(rcl_node_t* node, rclc_executor_t* executor) final
+    {
+        bool isSuccessful = false;
+
+        if ((nullptr == node) || (nullptr == executor))
+        {
+            LOG_ERROR("Node or executor are nullptr.");
+        }
+        else if (RCL_RET_OK != rclc_subscription_init_default(&m_subscriber, node, m_typeSupport, m_topicName.c_str()))
+        {
+            LOG_ERROR("Failed initialization of subscription.");
+        }
+        else if (RCL_RET_OK != rclc_executor_add_subscription_with_context(executor, &m_subscriber, &m_buffer,
+                                                                           &localCallback, this, ON_NEW_DATA))
+        {
+            LOG_ERROR("Failed to add subscription to executor.");
+        }
+        else
+        {
+            isSuccessful = true;
+        }
+
+        return isSuccessful;
+    }
+
+    /**
+     * Topic callback.
+     *
+     * @param[in] msgData   Incoming data from the topic message.
+     */
+    void topicCallback(const void* msgData) final
+    {
+        if (nullptr != msgData)
+        {
+            m_callback(reinterpret_cast<const messageType*>(msgData));
+        }
+    }
+
+private:
+    /**
+     * Callback for the subscription topic.
+     */
+    RosTopicCallback m_callback;
+
+    /**
+     * Buffer for the incomming messages.
+     */
+    messageType m_buffer;
+
+    /**
+     * Default Constructor.
+     * Not allowed.
+     */
+    Subscriber();
+};
+
+/******************************************************************************
+ * Functions
+ *****************************************************************************/
+
+/**
```

Comment:
move comment to prototype at beginning of file

Reply:
See comment on the declaration